src/Workspaces/Core/Portable/Shared/Utilities/AsyncBatchingWorkQueue.cs

```diff
@@ -128,6 +128,8 @@ public void AddWork(IEnumerable<TItem> items)
             }
         }
 
+        public async Task WaitUntilCurrentBatchCompletesAsync() => await _updateTask.ConfigureAwait(false);
```

Comment:
```suggestion
        public Task WaitUntilCurrentBatchCompletesAsync()
        {
            lock (_gate)
                return _updateTask;
        }
```

Reply:
this piece is subtle, and it's highly likely your code should work all the time.  but i'm going to be paranoid for safety.